### ISSUE TITLE
code declaration

### ISSUE THREAD

**Issue by celsofranssa** [conala-corpus/conala-corpus.github.io#2]
Is there any way to get the complete code (method declaration and method body) of each conala corpus code snippet?

For example, in addition to the json entries below, we would have the **code_declaration** attribute.

```
{
   "question_id": #####,
   "intent": "copying one file's contents to another",
   "rewritten_intent": "copy the content of source file to target file",
   "snippet": "shutil.copy (source_file, target_file)",
   "code_declaration": "def copy_file (source_file, target_file)"
}
```

Thank you.

**Comment by pcyin** [comment]
Hi, thanks a lot for your question! Having code declaration would be quite useful. Unfortunately we don't have such annotation. Perhaps the declaration could be inferred by performing static analysis on the snippet, which could be interesting future avenue.

**Comment by neubig** [comment]
Yes, this would be a great thing to have. One additional difficulty is that we would probably also need information about what the types and/or values the variables in the code can take. If this information was available this could be used for things like automatic testing of the outputs.

I'm not sure if we'll get around to this ourselves, but if somebody is interested in trying this I would be happy to give direction.

**Comment by celsofranssa** [comment]
Hello @pcyin ,
Could you give me more details on how to infer declaration by performing static analysis on the snippet?
